client/my-sites/checkout/composite-checkout/hooks/use-prepare-products-for-cart.ts

```diff
@@ -533,13 +551,17 @@ function createItemToAddToCart( {
 		createRequestCartProduct( {
 			product_id: productId,
 			product_slug: productSlug,
-		} )
+		} ),
+		{ isJetpackUserlessCheckout }
 	);
 }
 
-function addContextToProduct( product: RequestCartProduct ): RequestCartProduct {
+function addContextToProduct(
+	product: RequestCartProduct,
+	options?: Record< string, unknown >
+): RequestCartProduct {
 	return {
 		...product,
-		extra: { ...product.extra, context: 'calypstore' },
+		extra: Object.assign( { ...product.extra, context: 'calypstore' }, options ),
```

Comment:
Minor, but is there a benefit to using `Object.assign` here over just a spread?
```suggestion
		extra: { ...product.extra, context: 'calypstore', ...options },
```